pkg/pillar/cmd/usbmanager/usbevent.go

```diff
@@ -11,72 +11,115 @@ import (
 	"github.com/eshard/uevent"
 )
 
+type ueventListener struct {
+	isCancelled  atomic.Bool
+	ueventReader io.ReadCloser
+	usbdevices   map[string]*usbdevice
+	uc           *usbmanagerController
+}
+
 func (uc *usbmanagerController) listenUSBPorts() {
-	var isCancelled atomic.Bool
+	ul := ueventListener{
+		usbdevices: map[string]*usbdevice{},
+		uc:         uc,
+	}
+	ul.isCancelled.Store(false)
+	uc.listenUSBStopChan = make(chan struct{})
 
-	usbdevices := make(map[string]*usbdevice)
-	isCancelled.Store(false)
+	ul.waitForStop()
```

Comment:
This function name is a bit confusing since it doesn't wait at all. It spans a goroutine (which convents a message on the StopChan to  cancel/stop of the ueventReader.

So it might make more sense to do a "go ul.handleStop()" call.

Reply:
Fixed.